agent/src/beerocks/slave/platform_manager/platform_manager_thread.cpp

```diff
@@ -1312,7 +1312,7 @@ bool main_thread::handle_arp_monitor()
     sd = get_slave_socket_from_hostap_iface_name(strIfaceName);
 
     // Use the Backhaul Manager Slave as the default destination
-    if ((sd == nullptr) && ((sd = get_backhaul_socket()) == nullptr)) {
+    if ((!sd) && ((sd = get_backhaul_socket()))) {
```

Comment:
You changed the semantics here:

```suggestion
    if ((!sd) && (!(sd = get_backhaul_socket()))) {
```

Reply:
nice catch